lib/client/session.go

```diff
@@ -593,41 +621,57 @@ func (ns *NodeSession) pipeInOut(shell io.ReadWriteCloser) {
 			log.Errorf(err.Error())
 		}
 	}()
-	// copy from the local input to the remote shell:
-	go func() {
-		defer ns.closer.Close()
-		buf := make([]byte, 128)
-
-		stdin := ns.terminal.Stdin()
-		if ns.terminal.IsAttached() && ns.enableEscapeSequences {
-			stdin = escape.NewReader(stdin, ns.terminal.Stderr(), func(err error) {
-				switch err {
-				case escape.ErrDisconnect:
-					fmt.Fprintf(ns.terminal.Stderr(), "\r\n%v\r\n", err)
-				case escape.ErrTooMuchBufferedData:
-					fmt.Fprintf(ns.terminal.Stderr(), "\r\nerror: %v\r\nremote peer may be unreachable, check your connectivity\r\n", trace.Wrap(err))
-				default:
-					fmt.Fprintf(ns.terminal.Stderr(), "\r\nerror: %v\r\n", trace.Wrap(err))
+
+	switch mode {
+	case types.SessionObserverMode, types.SessionModeratorMode:
+		go func() {
+			defer ns.closer.Close()
+
+			handleNonPeerControls(mode, ns.terminal, func() {
+				_, err := sess.SendRequest(teleport.ForceTerminateRequest, true, nil)
+				if err != nil {
+					fmt.Printf("\n\rError while sending force termination request: %v\n\r", err.Error())
 				}
-				ns.closer.Close()
 			})
-		}
-		for {
-			n, err := stdin.Read(buf)
-			if err != nil {
-				fmt.Fprintf(ns.terminal.Stderr(), "\r\n%v\r\n", trace.Wrap(err))
-				return
+		}()
+	case types.SessionPeerMode:
+		// copy from the local input to the remote shell:
+		go func() {
+			defer ns.closer.Close()
+			buf := make([]byte, 128)
```

Comment:
Any particular reason for a buffer size of 128? Seems super small to me.

Reply:
No particular reason for that number specifically although it feels wasteful to allocate a large buffer since a read happens as often as possible. Typically every keystroke. The largest size event we're going to encounter is a larger copy paste. I bumped it up to 1KiB but I really don't have reasoning for allocating more.